ThermofluidStream/HeatExchangers/Internal/PartialDiscretizedHEX.mo

```diff
@@ -0,0 +1,104 @@
+  parameter SI.MassFlowRate m_flow_0_A=0 "Initial mass flow for side A" annotation (Dialog(
+      tab="Initialization",
+      group="Mass flow",
+      enable=initializeMassFlow));
+  parameter SI.MassFlowRate m_flow_0_B=0 "Initial mass flow for side B" annotation (Dialog(
+      tab="Initialization",
+      group="Mass flow",
+      enable=initializeMassFlow));
+  parameter Integer nCells=3 "Number of discretization elements";
+  parameter Modelica.Units.SI.Area A=10 "Conductive area of heat exchanger" annotation (Dialog(group="Heat transfer parameters"));
+  parameter Modelica.Units.SI.Volume V_Hex=0.001 "Volume for heat transfer calculation" annotation (Dialog(group="Heat transfer parameters"));
+  parameter SI.MassFlowRate m_flow_assert(max=0) = -dropOfCommons.m_flow_reg "Assertion threshold for negative massflows" annotation (Dialog(tab="Advanced"));
+  parameter Boolean enforce_global_energy_conservation=false "If true, exact global energy conservation is enforced by feeding back all energy stored locally back in the system" annotation (Dialog(tab="Advanced"));
+
+  //Parameterization of HEX Wall
+  parameter Modelica.Units.SI.CoefficientOfHeatTransfer k_wall=100 "Coefficient of heat transfer for pipe wall" annotation (Dialog(group="Heat transfer parameters"));
+
+  parameter Boolean calculate_efficiency=false "Enable calculation of efficiency";
+
+  SI.HeatFlowRate Q_flow_A=sum(thermalElementA.heatPort.Q_flow);
+  SI.HeatFlowRate Q_flow_B=sum(thermalElementB.heatPort.Q_flow);
+  SI.Mass M_A=sum(thermalElementA.M);
+  SI.Mass M_B=sum(thermalElementB.M);
+  SI.Energy deltaE_system=sum(thermalElementA.deltaE_system) + sum(thermalElementB.deltaE_system);
+
+  ThermofluidStream.HeatExchangers.Internal.DiscretizedHEXSummary summary "Summary record of Quantities";
+
+protected
+  parameter Modelica.Units.SI.ThermalConductance G=k_wall*A "Wall thermal conductance" annotation (Dialog(group="Wall parameters"));
+  parameter Integer nCellsParallel=2 "Number of discretization elements in parallel";
+  outer DropOfCommons dropOfCommons;
+
+  function efficiency = Internal.calculateEfficiency (redeclare package MediumA = MediumA, redeclare package MediumB = MediumB);
+
+public
+  Modelica.Thermal.HeatTransfer.Components.ThermalConductor thermalConductor[nCells](each G=G/nCells) annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=90,
+        origin={0,0})));
+
+  ConductionElementB thermalElementB[nCells] annotation (Placement(transformation(extent={{-10,90},{10,70}})));
+  ConductionElementA thermalElementA[nCells] annotation (Placement(transformation(extent={{10,-90},{-10,-70}})));
+
+  Interfaces.Inlet inletB(redeclare package Medium = MediumB) annotation (Placement(transformation(extent={{-110,70},{-90,90}}), iconTransformation(extent=if (nCellsParallel == 1) then {{-110,70},{-90,90}} else {{110,-90},{90,-70}})));
```

Comment:
If there are any neater methods to have the icon connectors in different positions for counterflow vs crossflow this should still still be implemented.

Reply:
Nothing that I know so far. 

But the side effect of your implemention is that the icon connectors of cross elements change their position for `nCells=1`, which is not the case for the original implementation of `DiscretizedCrossFlowHEX` and `DiscretizedCrossFlowHEX_FR`. Is this still the desired behavior? Optionally, a new protected parameter could be defined just for this.

Reply:
You are right, I will correct it to work the same as the NTU heat exchangers